include/RAJA/util/View.hpp

```diff
@@ -138,6 +143,45 @@ using TypedManagedArrayView = TypedViewBase<DataType,
 #endif
 
 
+template <typename ViewType,
+          typename AtomicPolicy = RAJA::auto_atomic>
+struct AtomicViewWrapper {
+  using Base = ViewType;
+  using DataPointer = typename Base::DataPointer;
+  using DataType = typename Base::DataType;
+  using AtomicType = RAJA::AtomicRef<DataType, AtomicPolicy>;
+
+  Base base_;
+
+  RAJA_INLINE
+  constexpr
+  explicit
+  AtomicViewWrapper(ViewType view)
+      : base_(view)
+  {
+  }
+
+  RAJA_INLINE void set_data(DataPointer data_ptr) { base_.set_data(data_ptr); }
+
+  template<typename ... ARGS>
+  RAJA_HOST_DEVICE
+  RAJA_INLINE
+  AtomicType operator()(ARGS &&... args) const
+  {
+    return AtomicType(&base_.operator()(args...));
```

Comment:
Prefer perfect forwarding the variadic args (since they were defined as a universal reference)  `std::forward<ARGS>(args)...`